src/inoreader_api.cpp

```diff
@@ -0,0 +1,300 @@
+#include <vector>
+#include <cstring>
+
+#include <inoreader_api.h>
+#include <config.h>
+#include <utils.h>
+#include <strprintf.h>
+
+#include <curl/curl.h>
+#include <json.h>
+
+#define INOREADER_LOGIN					"https://inoreader.com/accounts/ClientLogin"
+#define INOREADER_API_PREFIX				"https://inoreader.com/reader/api/0/"
+#define INOREADER_FEED_PREFIX				"https://inoreader.com/reader/atom/"
+
+#define INOREADER_SUBSCRIPTION_LIST		INOREADER_API_PREFIX "subscription/list"
+#define INOREADER_API_MARK_ALL_READ_URL	INOREADER_API_PREFIX "mark-all-as-read"
+#define INOREADER_API_EDIT_TAG_URL		INOREADER_API_PREFIX "edit-tag"
+#define INOREADER_API_TOKEN_URL			INOREADER_API_PREFIX "token"
+
+// for reference, see https://inoreader.com/developers
+
+namespace newsboat {
+
+inoreader_api::inoreader_api(configcontainer * c) : remote_api(c) {
+	// TODO
+}
+
+inoreader_api::~inoreader_api() {
+	// TODO
+}
+
+bool inoreader_api::authenticate() {
+	auth = retrieve_auth();
+	LOG(level::DEBUG, "inoreader_api::authenticate: Auth = %s", auth);
+	return auth != "";
+}
+
+static size_t my_write_data(void *buffer, size_t size, size_t nmemb, void *userp) {
+	std::string * pbuf = static_cast<std::string *>(userp);
+	pbuf->append(static_cast<const char *>(buffer), size * nmemb);
+	return size * nmemb;
+}
+
+std::string inoreader_api::retrieve_auth() {
+	CURL * handle = curl_easy_init();
+	credentials cred = get_credentials("inoreader", "inoreader");
+	if (cred.user.empty() || cred.pass.empty()) {
+		return "";
+	}
+
+	char * username = curl_easy_escape(handle, cred.user.c_str(), 0);
+	char * password = curl_easy_escape(handle, cred.pass.c_str(), 0);
+
+	std::string postcontent = strprintf::fmt("Email=%s&Passwd=%s", username, password);
+
+	curl_free(username);
+	curl_free(password);
+
+	std::string result;
+
+	curl_slist * list = NULL;
+	list = curl_slist_append(list, "AppId: 1000000588");
+	list = curl_slist_append(list, "AppKey: qRf9ZewWGwe630w61borYEtraeZMsMyS");
+
+	utils::set_common_curl_options(handle, cfg);
+	curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, my_write_data);
+	curl_easy_setopt(handle, CURLOPT_WRITEDATA, &result);
+	curl_easy_setopt(handle, CURLOPT_POSTFIELDS, postcontent.c_str());
+	curl_easy_setopt(handle, CURLOPT_URL, INOREADER_LOGIN);
+	curl_easy_perform(handle);
+	curl_easy_cleanup(handle);
+	curl_slist_free_all(list);
+
+	std::vector<std::string> lines = utils::tokenize(result);
+	for (auto line : lines) {
+		LOG(level::DEBUG, "inoreader_api::retrieve_auth: line = %s", line);
+		if (line.substr(0,5)=="Auth=") {
+			std::string auth = line.substr(5, line.length()-5);
```

Comment:
Just as with some of the things above, I just kept it this way based on the existing code because I could not find any contribution guidelines, but if you're open to improving the code, I'll certainly do it.

Reply:
Oh, I'm all for improving the code! Modernize everything you see :)

Opened #28 regarding guidelines; thanks for a prod. Never knew what I should write there, but seems like I should make an effort and figure it out if I want to see more PRs coming my way.